frontend/src/main.ts

```diff
@@ -55,10 +59,15 @@ async function getFirebaseConfig(): Promise<Object> {
 getFirebaseConfig().then(config => {
   firebase.initializeApp(config);
   firebase.auth().onAuthStateChanged(user => {
-    store.commit("user/setCurrentUser", user);
+    store.dispatch("user/handleUser", user);
   });
 
-  const unsubscribe: any = firebase.auth().onAuthStateChanged(() => {
+  const unsubscribe: any = firebase.auth().onAuthStateChanged(async user => {
+    const userRoles = user
+      ? (await user.getIdTokenResult()).claims.roles
+      : null;
+    await store.dispatch("user/updateUserRoles", userRoles);
```

Comment:
Aren't we doning the same in the `user` store module?

Reply:
This is needed for the initial page load, the one in the store is for auth state change when logging in and out.

Reply:
Can't we combine somehow?